src/mscorlib/src/System/Diagnostics/Eventing/XplatEventLogger.cs

```diff
@@ -79,6 +79,10 @@ private static string Serialize(ReadOnlyCollection<string> payloadName, ReadOnly
                 return String.Empty;
 
             Contract.Assert(payloadName.Count == payload.Count);
+            if(payloadName.Count != payload.Count)
```

Comment:
This function has "Contract.Assert(payloadName.Count == payload.Count);". Is this assert not valid and should be removed?